src/prefect_server/api/states.py

```diff
@@ -129,6 +130,12 @@ async def set_flow_run_state(
     if state.is_running() or state.is_submitted():
         await api.runs.update_flow_run_heartbeat(flow_run_id=flow_run_id)
 
+    # Set agent instance ID on flow run when submitted by agent
+    if state.is_submitted():
```

Comment:
```suggestion
    if state.is_submitted() and agent_instance_id:
```
no reason to hit the DB if no ID was provided